packages/synapse-react-client/src/components/ChallengeDataDownload/ChallengeDataDownload.tsx

```diff
@@ -0,0 +1,96 @@
+import React, { useCallback } from 'react'
+import AddCircleTwoToneIcon from '@mui/icons-material/AddCircleTwoTone'
+import { EntityType } from '@sage-bionetworks/synapse-types'
+import { EntityDetailsListSharedProps } from '../EntityFinder/details/EntityDetailsList'
+import { VersionSelectionType } from '../EntityFinder/VersionSelectionType'
+import { EntityFinderHeader } from '../EntityFinder/EntityFinderHeader'
+import { useEntitySelection } from '../EntityFinder/useEntitySelection'
+import { ChallengeEntityChildrenDetails } from './ChallengeEntityChildrenDetails'
+import { Box, Button } from '@mui/material'
+import { useAddFileBatchToDownloadList } from '../../synapse-queries'
+import { displayToast } from '../ToastMessage'
+import { useSynapseContext } from '../../utils'
+
+type ChallengeDataDownloadProps = {
+  parentContainerId: string
+}
+
+export function ChallengeDataDownload({
+  parentContainerId,
+}: ChallengeDataDownloadProps) {
+  const { downloadCartPageUrl } = useSynapseContext()
+  const selectMultiple = true
+  const [selectedEntities, toggleSelection] = useEntitySelection(selectMultiple)
+
+  const { mutate: addBatchToDownloadList } = useAddFileBatchToDownloadList({
+    onSuccess: () => {
+      displayToast(
+        'File(s) were successfully added to your Download Cart.',
+        'success',
+        {
+          primaryButtonConfig: {
+            text: 'View Download Cart',
+            onClick: () => (window.location.href = downloadCartPageUrl),
+          },
+        },
+      )
+    },
+    onError: e => {
+      displayToast(
+        `Unable to add the file to your Download Cart. ${e.reason}`,
+        'danger',
+      )
+    },
+  })
+
+  const isIdSelected = useCallback(
+    (entity: EntityFinderHeader) => {
+      return selectedEntities.has(entity.id)
+    },
+    [selectedEntities],
+  )
+
+  const onAddClick = useCallback(() => {
+    console.log({ selectedEntities })
+    const entities = selectedEntities.toArray().map(entity => {
+      return {
+        fileEntityId: entity[0],
+        versionNumber: entity[1],
+      }
+    })
+    addBatchToDownloadList(entities)
+  }, [selectedEntities, addBatchToDownloadList])
+
+  const sharedProps: EntityDetailsListSharedProps = {
+    versionSelection: VersionSelectionType.REQUIRED,
```

Comment:
Although version selection is not offered by this table, if we do not pass versionNumber to the add to cart service, it breaks the download cart page which displays `cannot call .toString() on undefined` in the error boundary.